### ISSUE TITLE
Let it snow? :snowflake: (Add first-party flake)

### ISSUE THREAD

**Issue by sbatial** [cafkafk/eza#72]
How are we feeling about adding a `flake.nix` and maybe try to move the dev and test-environment from Vagrant to nix as well?! (It seems like Vagrant and nix do not oppose each other... So we might just keep Vagrant but treat nix as the primary "environment control")

**Comment by cafkafk** [comment]
YES YES YES!

**Comment by cafkafk** [comment]
But... yes I wanted to do this, I just didn't get to it yet. If you wanna start it out I'm all for it!

**Comment by cafkafk** [comment]
I submitted a fix to the grid bug, so I don't think there's anything left that takes priority, gonna have a crack at this.

**Comment by sbatial** [comment]
Have fun.
I had created me a rust template once and would have started from there.
if you'd like to use it: https://gitlab.com/sbatial/nix-templates/-/blob/trunk/rust/flake.nix

**Comment by cafkafk** [comment]
>  I had created me a rust template once and would have started from there.

I ended up using a lot of this.

I am sure there is a lot more to be done on this front, but this starts us off.

**Comment by cafkafk** [comment]
One thing left is getting the extended tests running with nix, and not only vagrant.

**Comment by sbatial** [comment]
Very nice!

Any particular reason for all those empty lines in the flake.nix? :thinking:

**Comment by cafkafk** [comment]
> Very nice!
> 
> Any particular reason for all those empty lines in the flake.nix? thinking

No, I also dislike them, I just have seen a lot of people write them like that but if you also are not a fan I'll make it less ugly.

**Comment by sbatial** [comment]
Oh, ok. Interesting.

Yea, I think they can help legibility but too much of them kind of do the exact opposite.

I'll submit a PR with what I would have in mind
